habitat/core/benchmark.py

```diff
@@ -21,25 +21,95 @@ class Benchmark:
     r"""Benchmark for evaluating agents in environments.
     """
 
-    def __init__(self, config_paths: Optional[str] = None) -> None:
+    def __init__(
+        self, config_paths: Optional[str] = None, eval_remote=False
+    ) -> None:
         r"""..
 
         :param config_paths: file to be used for creating the environment
+        :param eval_remote: boolean indicating whether evaluation should be run remotely or locally
         """
         config_env = get_config(config_paths)
-        self._env = Env(config=config_env)
+        self._eval_remote = eval_remote
 
-    def evaluate(
+        if self._eval_remote is True:
+            self._env = None
+        else:
+            self._env = Env(config=config_env)
+
+    def remote_evaluate(
         self, agent: Agent, num_episodes: Optional[int] = None
-    ) -> Dict[str, float]:
-        r"""..
+    ):
+        # The modules imported below are specific to habitat-challenge remote evaluation.
+        # These modules are not part of the habitat-api repository.
+        import evaluation_pb2
+        import evaluation_pb2_grpc
+        import evalai_environment_habitat
+        import grpc
+        import pickle
+        import time
+
+        def pack_for_grpc(entity):
+            return pickle.dumps(entity)
+
+        def unpack_for_grpc(entity):
+            return pickle.loads(entity)
+
+        def remote_ep_over(stub):
```

Comment:
Maybe, a little more documentation or typing will help to understand and support this code easier later. That's not clear what is `stub`.